### PR TITLE
[FIX][2/2] fix step3-vl/deepseek-ocr image processor error

### PR DESCRIPTION
**Pull request by kousakawang**
## Motivation
Ref: https://github.com/sgl-project/sglang/issues/24699
Deepseek-OCR/OCR2  requests with JPEG images could fail with:

orig_width, orig_height = image.size

code here and following image-process code assuming image as PIL.image(cpu), while JpegGpu decoder output torch.Tensor on device.

## Modifications
Step1: Fix Step3-VL: https://github.com/sgl-project/sglang/pull/24701
Step2: Fix deepseek-ocr/ocr2

add  gpu process path for **tokenize_with_images**.
(fallback to cpu decoder also work, but with performance loss)

## Tests
Since DeepSeek-OCR is not a general VLM, only test with a script.
```python3
import base64
import requests
import io
import os
from PIL import Image
from openai import OpenAI
import sys
import time
import torch

CACHE_DIR = "image_cache"
os.makedirs(CACHE_DIR, exist_ok=True)

# --- Step 1: Parse argument and setup client ---
server_type = sys.argv[1] if len(sys.argv) > 1 else "sglang"
image_format = sys.argv[5].lower() if len(sys.argv) > 5 else "jpeg"
if image_format not in ("jpeg", "png"):
    print(f"Unsupported image format: {image_format}, use 'jpeg' or 'png'")
    sys.exit(1)
client = OpenAI(base_url="http://127.0.0.1:{}/v1".format(server_type), api_key="None")

image_urls = [
    "https://picsum.photos/seed/1/1280/720",
]

images_base64 = []

for idx, url in enumerate(image_urls):
    cache_path = os.path.join(CACHE_DIR, f"image_{idx}.{'jpg' if image_format == 'jpeg' else 'png'}")
    
    if os.path.exists(cache_path):
        print(f"load cache {idx+1}")
        img = Image.open(cache_path).convert("RGB")
    else:
        print(f"download {idx+1} from {url}")
        response = requests.get(url)
        img = Image.open(io.BytesIO(response.content)).convert("RGB")
        img.save(cache_path, format=image_format.upper())

    img = img.resize((int(sys.argv[2]), int(sys.argv[2])// 4 * 3 ))
    buffer = io.BytesIO()
    img.save(buffer, format=image_format.upper())
    base64_img = base64.b64encode(buffer.getvalue()).decode("utf-8")
    images_base64.append(base64_img)


messages = [
    {
        "role": "user",
        "content": [
            {"type": "text", "text": "Can you tell me what is in these images?" },
        ]
        + [{"type": "image_url", "image_url": {"url": f"data:image/{image_format};base64,{img_b64}"}} for img_b64 in images_base64] * int(sys.argv[3])
    }
]

response = client.chat.completions.create(
    model="/data01/models/DeepSeek-OCR-2",
    messages=messages,
    max_tokens=16,
    temperature=0.0,
)

print("Sending request to SGLang server....................................")
torch.cuda.synchronize()
loop_times = int(sys.argv[4])
start_time = time.time()
for i in range(loop_times):
    response_1 = client.chat.completions.create(
        model="/data01/models/DeepSeek-OCR-2",
        messages=messages,
        max_tokens=1,
        temperature=0.0,
    )
torch.cuda.synchronize()
end_time = time.time()
print("infer cost {} ms".format((end_time - start_time) * 1000 / loop_times))

print("\n=== Model Response ===")
print(response.choices[0].message.content)

```
test with
> python3 test_deepseek_ocr.py  8080 512 1 1 jpeg

**original output (test with with png or fallback to cpu)**
infer cost 106.23104572296143 ms

=== Model Response ===
 [Image description]The image displays a photograph of the Eiffel Tower.

**pr output**
infer cost 88.64872455596924 ms

=== Model Response ===
 [Image description]The image displays a photograph of the Eiffel Tower.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
3. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
4. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
5. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #26294055715](https://github.com/sgl-project/sglang/actions/runs/26294055715)
Latest PR Test (Extra): :x: [Run #26294055211](https://github.com/sgl-project/sglang/actions/runs/26294055211)

### CODE REVIEW

**Comment by yuan-luo** [comment]
Could you please address gemini's comments?

**Review by yhyang201 — APPROVED** [review]
